suite_healthcheck_test.go

```diff
@@ -34,14 +37,21 @@ func (f *TestSuite) TestHealthCheck() {
 	ctx, cancel := context.WithTimeout(f.ctx, contextTimeout)
 	defer cancel()
 
+	hcRequest := &grpc_health_v1.HealthCheckRequest{
+		Service: "networkservice.NetworkService",
+	}
+
 	healthClient := grpc_health_v1.NewHealthClient(f.sutCC)
-	healthResponse, err := healthClient.Check(ctx,
-		&grpc_health_v1.HealthCheckRequest{
-			Service: "networkservice.NetworkService",
-		},
-		grpc.WaitForReady(true),
-	)
+	healthResponse, err := healthClient.Check(ctx, hcRequest, grpc.WaitForReady(true))
 	f.NoError(err)
 	f.Require().NotNil(healthResponse)
 	f.Equal(grpc_health_v1.HealthCheckResponse_SERVING, healthResponse.Status)
+
+	f.Require().Eventually(func() bool {
+		healthResponse, err = healthClient.Check(ctx, hcRequest)
+		return err != nil
+	}, f.idleTimeout+time.Second, time.Millisecond*50)
+	f.Require().Nil(healthResponse)
+	f.Require().Error(err)
+	f.Require().Equal(codes.Unavailable, grpcutils.UnwrapCode(err))
```

Comment:
I think it is not good to shut down the endpoint in the suite because it can break the next tests in the suite. 
Can we create a separate test for onidle case?

Reply:
I tried to make a separate test, and it had hanging-forever issues, probably because current suite setup code doesn't account for the possibility that it can be run several times.

Currently suite only has 1 test. I tried to think of which tests can be added to it, and didn't come up with anything, so I figured out I can just make this test terminating.

I'll try to modify the setup code we are expecting the we will add more tests to this suite in the future.

Reply:
done